axlearn/experiments/text/gpt/c4_trainer.py

```diff
@@ -149,4 +152,50 @@ def make_single_host_config(base_config_name: str) -> SpmdTrainer.Config:
                 config_map[f"{config_name}-single-host"] = functools.partial(
                     make_single_host_config, config_name
                 )
+
+            if model_size == "test":
+
+                def make_simple_test_config(
+                    base_config_name: str,
+                    batch_size: int = 32,
+                    eval_every_n_steps: int = 2000,
+                    max_step: int = 3000,
+                ) -> SpmdTrainer.Config:
+                    """Make a variant of fuji-test of the test config
+                    that terminates early and saves checkpoints frequently.
+
+                    Args:
+                        base_config_name: The test config name.
+                        batch_size: The global batch size for training inputs.
+                        eval_every_n_steps: How often to run evaluation.
+                        max_step: The maximum number of training steps.
+
+                    Returns:
+                        A trainer config that can run for a short period of time.
+                    """
+
+                    # pytype: disable=annotation-type-mismatch
+                    cfg: SpmdTrainer.Config = config_map[base_config_name]().clone()
+                    # pytype: enable=annotation-type-mismatch
+
+                    cfg.input.batcher.global_batch_size = batch_size
+                    for evaler in cfg.evalers.values():
+                        evaler.input.batcher.global_batch_size = batch_size
+                        evaler.set(
+                            eval_policy=config_for_function(eval_every_n_steps_policy).set(
+                                n=eval_every_n_steps
+                            )
+                        )
+                    cfg.max_step = max_step
+                    cfg.mesh_shape = mesh_shape_from_axes(data=-1, fsdp=4)
+                    cfg.summary_writer.write_every_n_steps = eval_every_n_steps
+                    cfg.checkpointer.save_policy = config_for_function(every_n_steps_policy).set(
+                        n=eval_every_n_steps
```

Comment:
Is it possible to save checkpointer more frequently than eval? Something like save ckpt every 500 steps, eval every 1500 steps. This allows us to identify issues separately if the job hangs

Reply:
Thanks for the feedback! Added the custom policy to save checkpoints more often to differentiate from eval steps